### ISSUE TITLE
filter woocommerce_available_payment_gateways is causing admin to break

### ISSUE THREAD

**Issue by Pikkupomo** [maksuturva/woocommerce_payment_module#23]
We have a custom code wrapper for Woocommerce `get_available_payment_gateways()`:

```php
public static function get_enabled_payment_gateways()
    {
        $enabled_gateways = [];
        $gateways = WC()->payment_gateways()->get_available_payment_gateways();

        if ($gateways) {
            foreach ($gateways as $gateway) {
                if ($gateway->enabled == 'yes') {
                    $enabled_gateways[] = $gateway;
                }
            }
        }

        return $enabled_gateways;
    }
```

this code is being called in admin and it throws `WC_Gateway_Maksuturva_Exception` and breaks the entire admin. The underlying reason is that your code tries to access API even though the payment method has not been configured in any way.

**Problems**
1) Payment gateway is enabled by default
2) API is being used in admin even though there is no actual checkout transaction going on
3) Payment gateway is acting like it's enabled even though it's not

Core issue is in the method `private function get_available_payment_methods($price)`
which calls API in a pretty blind manner:
```php
$api = new WC_Svea_Api_Request_Handler($this->gateway);

$result_methods = $api->post(
    self::ROUTE_RETRIEVE_AVAILABLE_PAYMENT_METHODS,
    $post_fields
);
```

secondary issue can be found in `payment_gateway_disable_empty`:
```php
$payment_methods = $this->payment_method_select->get_payment_type_payment_methods(
    $payment_method_type,
    WC_Payment_Gateway::get_order_total()
);
``` 

**Suggestion**
Wrap the `add_filter` to work only if it's being called in frontend:

```php
if (!is_admin()) {
    add_filter('woocommerce_available_payment_gateways', [$this, 'payment_gateway_disable_empty']);
}
```

I'm also wondering if cron causes this to fail as well?

This is the first payment gateway doing this, so I wonder, can we expect a fix related to this at some point? I would not like to keep modified version of your plugin in use :)

**Comment by Svea-Maintainer** [comment]
Thanks for the comprehensive report... 

Have you made a fixed version of the payment module? If you want, you can make a pull request, and we will take it to the main branch after testing. Otherwise, the repair will take a bit because we will probably order it from a subcontractor.

But we'll take a closer look soon.

**Comment by Svea-Maintainer** [comment]
And if this fix below is sufficient, we can make that quite fast.
`if (!is_admin()) {
    add_filter('woocommerce_available_payment_gateways', [$this, 'payment_gateway_disable_empty']);
}`
some testing needs to be done still.

**Comment by Svea-Maintainer** [comment]
Fixed as proposed in version 2.1.7
